src/helpers/counterBuilder.js

```diff
@@ -0,0 +1,229 @@
+    const from = (this.props.from === undefined) ? startTime : this.props.from
+    const timeDiff = (this.props.seconds === undefined)
+      ? (this.props.to - from)
+      : (this.props.seconds * 1000)
+    this.state = {
+      ...this.state,
+      startTime,
+      from,
+      timeDiff,
+      initialTimeDiff: timeDiff
+    }
+  }
+
+  /**
+   * Gets an array of periods to create segments for.
+   */
+  buildPeriods () {
+    const periods = PERIODS.slice(
+      PERIODS.indexOf(this.props.maxPeriod),
+      PERIODS.indexOf(this.props.minPeriod) + 1
+    )
+    this.state = {
+      ...this.state,
+      periods
+    }
+  }
+
+  /**
+   * Builds minDigits and maxDigits state properties.
+   */
+  buildDigitLimits () {
+    var [minDigits, maxDigits] = this.getDigitLimitsFromProps()
+    this.state = {
+      ...this.state,
+      ...this.normalizeDigitLimits(minDigits, maxDigits)
+    }
+  }
+
+  /**
+   * Gets minDigits and maxDigits from props specified by the user.
+   * @return {array}
+   */
+  getDigitLimitsFromProps () {
+    var minDigits = {}
+    var maxDigits = {}
+
+    if (typeof this.props.minDigits === 'number') {
+      for (let period of this.state.periods) minDigits[period] = this.props.minDigits
+    } else if (typeof this.props.minDigits === 'object') {
+      minDigits = this.props.minDigits
+    }
+    if (typeof this.props.maxDigits === 'number') {
+      for (let period of this.state.periods) maxDigits[period] = this.props.maxDigits
+    } else if (typeof this.props.maxDigits === 'object') {
+      maxDigits = this.props.maxDigits
+    }
+    return [minDigits, maxDigits]
+  }
+
+  /**
+   * Calculates minimum number of digits for current radix and given period.
+   * @param {string} period - period to calculate minDigits for
+   * @return {number}
+   */
+  getMinSegmentSize (period) {
+    const maxValue = this.getPeriodMaxValue(period)
+    return (maxValue).toString(this.props.radix).length
+  }
+
+  /**
+   * Normalizes minDigits and maxDigits and sets default values where user did not specify them.
+   * @param {number|object} minDigits
+   * @param {number|object} maxDigits
+   * @return {object}
+   */
+  normalizeDigitLimits (minDigits, maxDigits) {
+    var normalizedMinDigits = {}
+    var normalizedMaxDigits = {}
+
+    for (let period of this.state.periods) {
+      const minSegmentSize = this.getMinSegmentSize(period);
+      [normalizedMinDigits[period], normalizedMaxDigits[period]] = this.normalizeDigitLimitsForPeriod(
+        minDigits[period], maxDigits[period], minSegmentSize
+      )
+    }
+    return {
+      minDigits: normalizedMinDigits,
+      maxDigits: normalizedMaxDigits
+    }
+  }
+
+  normalizeDigitLimitsForPeriod (min, max, minSegmentSize) {
+    const minProvided = min !== undefined
+    const maxProvided = max !== undefined
+    function throwError () {
+      throw new Error(`conflict: minDigits (${min}) > maxDigits (${max})`)
+    }
+
+    if (minProvided && max) {
```

Comment:
Right. If `maxDigits` is set to zero, we should keep it that way no mater what `minDigits` is set to. Of course, this will be documented in the README eventually.